src/tap.cpp

```diff
@@ -41,27 +48,135 @@ TAP::TAP(QObject *parent) : QObject(parent) {
         close(mFd);
         exit(1);
     }
+    if (ioctl(mFd, TUNSETOFFLOAD, 0U) < 0) {
+        qCritical() << "Failed to set the TAP offload";
+    }
+
+    mIfaceName = ifr.ifr_name;
+    qDebug() << "TAP interface created:" << mIfaceName;
 
-    QString ifaceName = ifr.ifr_name;
-    qDebug() << "TAP interface created:" << ifaceName;
+    iffReset(244);
 
-    // Bring the interface up
-    if (QProcess::execute("ip", {"link", "set", "dev", ifaceName, "up"}) != 0) {
-        qCritical() << "Failed to bring up TAP interface";
+    char discard[1500];
+    auto drained = read(mFd, discard, 1500);
+    qDebug() << "Drained" << drained << "bytes";
+
+    // Poll the interface file descriptor
+    mNotifier = new QSocketNotifier(mFd, QSocketNotifier::Read, this);
+    QObject::connect(mNotifier, &QSocketNotifier::activated, this, &TAP::fdActivated);
+}
+
+void TAP::iffReset(int mtu) {
+    struct ifreq ifr = {};
+    auto sock = socket(PF_INET, SOCK_DGRAM, 0);
+    if (sock < 0) {
+        qCritical() << "Failed to create a socket";
         exit(1);
     }
-    qDebug() << "TAP interface" << ifaceName << "is up";
-    
-    // Configure routing tables
-    if (QProcess::execute("ip", {"route", "add", "default", "dev", ifaceName}) != 0) {
-        qCritical() << "Failed to configure routing table";
+    strncpy(ifr.ifr_name, mIfaceName.toLocal8Bit().data(), sizeof(ifr.ifr_name));
+    if (ioctl(sock, SIOCGIFFLAGS, &ifr) < 0) {
+        qCritical() << "Failed to get IFFLAGS";
+        exit(1);
+    }
+    if (ifr.ifr_flags & IFF_UP) {
+        ifr.ifr_flags &= ~IFF_UP;
+        if (ioctl(sock, SIOCSIFFLAGS, &ifr) < 0) {
+            qCritical() << "Failed to set IFFLAGS";
+            exit(1);
+        }
+    }
+    if (mtu < 0) {
+        qDebug() << "Interface down";
+        close(sock);
+        return;
+    }
+    ifr.ifr_flags |= IFF_UP;
+    if (ioctl(sock, SIOCSIFFLAGS, &ifr) < 0) {
+        qCritical() << "Failed to set IFFLAGS";
         exit(1);
     }
-    qDebug() << "Routing table configured to send all traffic to" << ifaceName;
 
-    // Poll the interface file descriptor
-    mNotifier = new QSocketNotifier(mFd, QSocketNotifier::Read, this);
-    QObject::connect(mNotifier, &QSocketNotifier::activated, this, &TAP::fdActivated);
+    ifr.ifr_mtu = mtu - MY_ETH_TAP_HARD_FRAME_SIZE;
```

Comment:
Let's reuse the mtuChanged() function here to avoid code duplication and chances that implementations drift over time :)